dmitrii-ubskii/typedb
# Remove vaticle factory tracing and upgrade build with latest grpc rules

## What is the goal of this PR?

Upgrade to the latest GRPC build rules, which induces deleting factory-tracing and packaging TypeDB's dependencies without maven coordinate-style naming.

## What are the changes implemented in this PR?

* Upgrade to the latest protocol and dependencies versions

The new grpc build rules don't pull their dependencies via maven coordinate anymore, which means our maven assembly rules fail. This necessitates two changes:

* Remove `factory-tracing` completely - we will rewrite & reintegrate it into 3.0 in Rust when we get to it, with async support. 
* Remove TypeDB's packaging of JARs using maven naming conventions

These changes mean we no longer have to deploy any maven artifacts that use the new protocol build rules.
